x/fungible/keeper/zrc20_cosmos_coins_mapping.go

```diff
@@ -0,0 +1,191 @@
+package keeper
+
+import (
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/zeta-chain/node/pkg/crypto"
+	fungibletypes "github.com/zeta-chain/node/x/fungible/types"
+)
+
+// LockZRC20 locks ZRC20 tokens in the specified address
+// The caller must have approved the locker contract to spend the amount of ZRC20 tokens.
+// Warning: This function does not mint cosmos coins, if the depositor needs to be rewarded
+// it has to be implemented by the caller of this function.
+func (k Keeper) LockZRC20(
+	ctx sdk.Context,
+	zrc20ABI *abi.ABI,
+	zrc20Address, owner, locker common.Address,
+	amount *big.Int,
+) error {
+	// owner is the EOA owner of the ZRC20 tokens.
+	// locker is the address that will lock the ZRC20 tokens, i.e: bank precompile.
+	if err := k.CheckZRC20Allowance(ctx, zrc20ABI, owner, locker, zrc20Address, amount); err != nil {
+		return err
```

Comment:
Can we use `errors.Wrap` for all these errors?

Reply:
Wrapped all the non-typed errors in 6adb5ec9bdbdf6b1239764fda7509a62bdf27ecb